hub/api/tests/test_api.py

```diff
@@ -464,3 +465,32 @@ def test_hub_cloud_dataset():
         np.testing.assert_array_equal(ds.image[i].numpy(), i * np.ones((100, 100)))
 
     ds.delete()
+
+
+def test_iter_perf(memory_ds: Dataset):
```

Comment:
```suggestion
def test_binary_search_iteration_performance(memory_ds: Dataset):
```

Reply:
We are actually checking that we are not binary searching (few bin searches are still made by shape and byte pos encoder), not the performance of the bin search itself.